qendpoint-store/src/test/java/com/the_qa_company/qendpoint/TempTest.java

```diff
@@ -0,0 +1,181 @@
+package com.the_qa_company.qendpoint;
+
+import com.the_qa_company.qendpoint.compiler.CompiledSail;
+import com.the_qa_company.qendpoint.compiler.SparqlRepository;
+import com.the_qa_company.qendpoint.core.hdt.HDT;
+import com.the_qa_company.qendpoint.core.hdt.HDTManager;
+import com.the_qa_company.qendpoint.core.options.HDTOptions;
+import com.the_qa_company.qendpoint.core.options.HDTOptionsKeys;
+import com.the_qa_company.qendpoint.core.triples.TripleString;
+import com.the_qa_company.qendpoint.store.EndpointFiles;
+import com.the_qa_company.qendpoint.store.Utility;
+import org.apache.commons.lang3.time.StopWatch;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.query.TupleQuery;
+import org.eclipse.rdf4j.query.explanation.Explanation;
+import org.eclipse.rdf4j.repository.sail.SailRepositoryConnection;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class TempTest {
+
+	@Rule
+	public TemporaryFolder tempDir = TemporaryFolder.builder().assureDeletion().build();
+	private SparqlRepository repository;
+
+
+	@Before
+	public void setupRepo() throws IOException {
+		Path root = tempDir.newFolder().toPath();
+		ClassLoader loader = getClass().getClassLoader();
+		String filename = "2018_complete.nt";
+
+		Path hdtstore = root.resolve("hdt-store");
+		Path locationNative = root.resolve("native");
+
+		Files.createDirectories(hdtstore);
+		Files.createDirectories(locationNative);
+
+		String indexName = "index.hdt";
+
+		HDTOptions options = HDTOptions.of(
+				// disable the default index (to use the custom indexes)
+				HDTOptionsKeys.BITMAPTRIPLES_INDEX_NO_FOQ, true,
+				// set the custom indexes we want
+				HDTOptionsKeys.BITMAPTRIPLES_INDEX_OTHERS, "sop,ops,osp,pso,pos");
+
+
+		try (HDT hdt = HDTManager.generateHDT(new Iterator<>() {
+			@Override
+			public boolean hasNext() {
+				return false;
+			}
+
+			@Override
+			public TripleString next() {
+				return null;
+			}
+		}, Utility.EXAMPLE_NAMESPACE, options, null)) {
+			hdt.saveToHDT(hdtstore.resolve(indexName).toAbsolutePath().toString(), null);
+		} catch (Error | RuntimeException e) {
+			throw e;
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+
+		repository = CompiledSail.compiler().withEndpointFiles(new EndpointFiles(locationNative, hdtstore, indexName))
+				.compileToSparqlRepository();
+		try (InputStream is = new BufferedInputStream(Objects.requireNonNull(loader.getResourceAsStream(filename),
+				filename + " doesn't exist"))) {
+			repository.loadFile(is, filename);
+		}
+	}
```

Comment:
@ate47 I'm using this code to set up a repo with the file from https://github.com/the-qa-company/qEndpoint/discussions/413 

The query statistics seems to always be returning 0. Do you know why?

Reply:
Can you detail this a bit more. I loaded the dataset and I was querying one triple pattern and I see that the cardinality is non zero:
<img width="1141" alt="Screenshot 2024-01-22 at 22 52 51" src="https://github.com/the-qa-company/qEndpoint/assets/10883585/d59d4cbb-f4b9-4ae2-981a-b1b457f718ad">

Reply:
Try adding a break point on that line and then debugging the TempTest test() I added.

Reply:
Don't forget to add the file with the data to this path: qendpoint-store/src/test/resources/2018_complete.nt

Reply:
It's a race condition because a merge was triggered in the middle of the file loading

Reply:
Do you know how I can wait for the merge to complete?

Reply:
Btw. In my PR https://github.com/eclipse-rdf4j/rdf4j/pull/4879 I managed to get the QueryJoinOptimizer to work better when the statistics are inaccurately returning 0.0 by adding 5.0 to the cost of all operations and also penalising cartesian joins more heavily.